cpp/models/lct_secir/parameters_io.h

```diff
@@ -44,34 +47,48 @@ namespace details
 { // Use namespace to hide functions that are not intended to be used outside this file.
 
 /**
-* @brief Processes one entry of an RKI data set for the definition of an initial value vector for an LCT model.
+ * @brief Check whether the EntryType is age resolved and has a member variable age_group_names.
+ * @tparam The type to check for the existence of the member variable age_group_names.
+ */
+template <class EntryType>
+using has_age_member_t = decltype(EntryType::age_group_names);
+
+template <class EntryType>
+using is_age_resolved_entry_type = is_expression_valid<has_age_member_t, EntryType>;
```

Comment:
It is a good idea to use this pattern to test for a member, but I have a some small concerns.

You check here whether age_group_names exists, but use age_group in line 409 instead. To me this indicates that you do not check for (all) the member(s) you actually intend to use; or that maybe you expect a specific type of Entry in 408 instead, in which case you could use `std::is_same_v<EntryType, OtherType>` directly.

Other than that, words like "has" or "is" are commonly used to indicate boolean expressions, or with types something that has a boolean member value (e.g. std::true_type or std::false_type, or some type inheriting either). So "has_age_member_t" could be misleading. For types intended for use with "is_expression_valid", we usually name them as "member-name_expr_t". For an example, see the bottom of flow_model.h.

Reply:
Actually, i expect specific types, one for age resolved and one for the case without age resolution. Therefore, is_same_v is more accurate.